apps/mobile/src/hooks/use-copy-to-clipboard.ts

```diff
@@ -0,0 +1,34 @@
+import { useCallback, useState } from 'react';
+
+import * as Clipboard from 'expo-clipboard';
+
+const defaultSuccessStateDuration = 2000;
+
+export type UseCopyToClipboardStatus = 'idle' | 'success' | 'error';
+
+interface UseCopyToClipboardOptions {
+  successStateDuration?: number;
+}
+
+export function useCopyToClipboard(value: string, options: UseCopyToClipboardOptions = {}) {
+  const { successStateDuration = defaultSuccessStateDuration } = options;
+  const [status, setStatus] = useState<UseCopyToClipboardStatus>('idle');
+
+  const onCopy = useCallback(async () => {
+    if (value.trim() === '') return;
+
+    try {
+      await Clipboard.setStringAsync(value);
+      setStatus('success');
+    } catch {
+      setStatus('error');
+    }
+  }, [value]);
+
+  setTimeout(() => setStatus('idle'), successStateDuration);
```

Comment:
Unsafe use of `setTimeout`. If component dismounts before this period ends, react will throw an error.

Reply:
The warning [was removed in React 18](https://react.dev/blog/2022/03/08/react-18-upgrade-guide#react), this is a negligible memory leak, safe to ignore. 
But thanks for catching this–should be in `finally`, no idea why I placed into the render